internal/stores/objects.go

```diff
@@ -304,9 +305,18 @@ type (
 		Model
 		DBSliceID uint `gorm:"index"`
 
-		Key       []byte `gorm:"unique;NOT NULL"` // json string
-		MinShards uint8
-		Shards    []dbSector `gorm:"many2many:sector_slabs;constraint:OnDelete:CASCADE"` // CASCADE to delete shards too
+		Key         []byte    `gorm:"unique;NOT NULL"` // json string
+		LastFailure time.Time `gorm:"index"`
+		MinShards   uint8
+		Shards      []dbShard `gorm:"constraint:OnDelete:CASCADE"` // CASCADE to delete shards too
+	}
+
+	// dbShard is a join table between dbSlab and dbSector.
+	dbShard struct {
```

Comment:
`nit` while I like the name I would usually name join tables using the table names involved, e.g. `dbSlabSector` - we can definitely keep calling it shards though just figured I'd mention it

Reply:
I don't care too much. I did mostly choose that name over the one I had before because it's no longer a textbook gorm join table. Because it got its own ID and the other id fields are not primary keys anymore. 

That's because the regular join table with `many2many` won't allow for a slab to contain the same sector multiple times which causes issues. So I build this manual join table to allow for that edge case.